src/commands/staking/delegate/mod.rs

```diff
@@ -0,0 +1,109 @@
+use strum::{EnumDiscriminants, EnumIter, EnumMessage};
+
+mod deposit;
+mod deposit_and_stake;
+mod stake;
+mod stake_all;
+mod unstake;
+mod unstake_all;
+pub mod view_balance;
+mod withdraw;
+mod withdraw_all;
+
+#[derive(Debug, Clone, interactive_clap::InteractiveClap)]
+#[interactive_clap(input_context = crate::GlobalContext)]
+#[interactive_clap(output_context = DelegateStakeContext)]
+pub struct DelegateStake {
+    #[interactive_clap(skip_default_input_arg)]
+    /// Enter the account that you want to manage delegated stake for:
+    account_id: crate::types::account_id::AccountId,
+    #[interactive_clap(subcommand)]
+    delegate_stake_command: DelegateStakingCommand,
+}
+
+#[derive(Debug, Clone)]
+pub struct DelegateStakeContext {
+    global_context: crate::GlobalContext,
+    account_id: near_primitives::types::AccountId,
+}
+
+impl DelegateStakeContext {
+    pub fn from_previous_context(
+        previous_context: crate::GlobalContext,
+        scope: &<DelegateStake as interactive_clap::ToInteractiveClapContextScope>::InteractiveClapContextScope,
+    ) -> color_eyre::eyre::Result<Self> {
+        if !crate::common::is_used_delegated_validator_list_exist(
+            &previous_context.config.credentials_home_dir,
+        ) {
+            crate::common::create_used_delegated_validator_list(&previous_context.config)?;
+        }
+        Ok(Self {
+            global_context: previous_context,
+            account_id: scope.account_id.clone().into(),
+        })
+    }
+}
+
+impl DelegateStake {
+    pub fn input_account_id(
+        context: &crate::GlobalContext,
+    ) -> color_eyre::eyre::Result<Option<crate::types::account_id::AccountId>> {
+        crate::common::input_non_signer_account_id_from_used_account_list(
+            &context.config.credentials_home_dir,
+            "Enter the account that you want to manage delegated stake for:",
+        )
+    }
+}
+
+#[derive(Debug, EnumDiscriminants, Clone, interactive_clap::InteractiveClap)]
+#[interactive_clap(context = DelegateStakeContext)]
+#[strum_discriminants(derive(EnumMessage, EnumIter))]
+#[non_exhaustive]
+/// Select actions with delegated staking:
+pub enum DelegateStakingCommand {
```

Comment:
Let's rename to `StakeDelegationCommand` or `StakeDelegationCmd`:

```suggestion
pub enum StakeDelegationCommand {
```

Reply:
Done